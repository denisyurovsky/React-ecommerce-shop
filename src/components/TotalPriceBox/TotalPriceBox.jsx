import { Stack } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';

import { formatPrice } from '../../helpers/formatData';

import DiscountBox from './DiscountBox/DiscountBox';

import styles from './TotalPriceBox.module.scss';

const TotalPriceBox = ({ cart }) => {
  const { totalPrice, totalDiscountPrice, sellersDiscount, personalDiscount } =
    cart;

  const shouldDisplayDiscount =
    totalDiscountPrice !== totalPrice && totalDiscountPrice !== 0;

  if (!shouldDisplayDiscount) {
    <p data-testid="totalPrice" className={styles.totalPrice}>
      Total: {formatPrice(totalPrice)}
    </p>;
  }

  return (
    <Stack className={styles.priceBox}>
      <DiscountBox
        sellersDiscount={sellersDiscount}
        personalDiscount={personalDiscount}
      />
      <Stack
        direction="row"
        data-testid="totalPrice"
        className={styles.totalPrice}
      >
        <p>Total:&nbsp;</p>
        <p>{formatPrice(totalDiscountPrice)}</p>
      </Stack>
    </Stack>
  );
};

TotalPriceBox.propTypes = {
  cart: PropTypes.shape({
    totalPrice: PropTypes.number,
    totalDiscountPrice: PropTypes.number,
    sellersDiscount: PropTypes.number,
    personalDiscount: PropTypes.number,
  }),
};

export default TotalPriceBox;
