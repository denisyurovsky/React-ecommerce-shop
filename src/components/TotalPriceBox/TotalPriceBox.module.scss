.priceBox {
  padding: 10px;
  padding-bottom: 0;
  margin-right: 20px;
  max-width: 270px;
}

.totalPrice {
  font-size: 18px;
  font-weight: bold;
  justify-content: space-between;

  & p {
    margin-bottom: 0;
  }
}

.totalDiscountPrice {
  font-size: 20px;
  font-weight: bold;
}

.savedMoney {
  justify-content: space-between;
  font-size: 18px;
}
