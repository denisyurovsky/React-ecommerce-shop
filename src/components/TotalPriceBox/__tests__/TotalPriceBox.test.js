import { render } from '@testing-library/react';
import React from 'react';

import TotalPriceBox from '../TotalPriceBox';

const cartWithDiscount = {
  totalPrice: 1000,
  totalDiscountPrice: 500,
  sellersDiscount: 200,
  personalDiscount: 300,
};

const cartWithoutDiscount = {
  totalPrice: 1000,
  totalDiscountPrice: 1000,
  sellersDiscount: 0,
  personalDiscount: 0,
};

describe('TotalPriceBox render correct snapshots', () => {
  it('should render a valid snapshot with discounts', () => {
    const { asFragment } = render(<TotalPriceBox cart={cartWithDiscount} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a valid snapshot without discounts', () => {
    const { asFragment } = render(<TotalPriceBox cart={cartWithoutDiscount} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
