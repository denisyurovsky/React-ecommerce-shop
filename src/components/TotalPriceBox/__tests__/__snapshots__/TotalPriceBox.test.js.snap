// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalPriceBox render correct snapshots should render a valid snapshot with discounts 1`] = `
<DocumentFragment>
  <div
    class="priceBox css-nen11g-MuiStack-root"
  >
    <div
      class="container MuiBox-root css-0"
    >
      <p
        class="title"
      >
        Discounts:
      </p>
      <table
        class="table"
      >
        <tbody>
          <tr
            class="row"
          >
            <td>
              General discounts:
            </td>
            <td>
              - 200 $
            </td>
          </tr>
          <tr
            class="row"
          >
            <td>
              Personal discounts:
            </td>
            <td>
              - 300 $
            </td>
          </tr>
          <tr
            class="row"
          >
            <td>
              Total discounts:
            </td>
            <td>
              - 500 $
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="totalPrice css-m69qwo-MuiStack-root"
      data-testid="totalPrice"
    >
      <p>
        Total: 
      </p>
      <p>
        500 $
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TotalPriceBox render correct snapshots should render a valid snapshot without discounts 1`] = `
<DocumentFragment>
  <div
    class="priceBox css-nen11g-MuiStack-root"
  >
    <div
      class="totalPrice css-m69qwo-MuiStack-root"
      data-testid="totalPrice"
    >
      <p>
        Total: 
      </p>
      <p>
        1 000 $
      </p>
    </div>
  </div>
</DocumentFragment>
`;
